File: 3771.total-score-of-dungeon-runs.py
#
# @lc app=leetcode id=3771 lang=python3
#
# [3771] Total Score of Dungeon Runs
#

# @lc code=start
class Solution:
    def totalScore(self, hp: int, damage: List[int], requirement: List[int]) -> int:
        n = len(damage)
        total_score = 0
        
        # Iterate over each possible starting room index
        for start in range(n):
            current_hp = hp
            score = 0
            
            # Simulate moving through rooms from 'start' to 'end'
            for i in range(start, n):
                current_hp -= damage[i]
                if current_hp >= requirement[i]:
                    score += 1
            
            total_score += score
        
        return total_score
# @lc code=end

File: 3771.total-score-of-dungeon-runs.cpp
#
# @lc app=leetcode id=3771 lang=cpp
#
# [3771] Total Score of Dungeon Runs
#
# @lc code=start
class Solution {
public:
    long long totalScore(int hp, vector<int>& damage, vector<int>& requirement) {
        int n = damage.size();
        long long total_score = 0;
        for (int j = 0; j < n; ++j) {
            int current_hp = hp;
            int score_j = 0;
            for (int i = j; i < n; ++i) {
                current_hp -= damage[i];
                if (current_hp >= requirement[i]) {
                    ++score_j;
                }
            }
            total_score += score_j;
        }
        return total_score;
    }
};
# @lc code=end